tests/conftest.py

```diff
@@ -19,72 +19,32 @@
 import pytest
 
 from ludwig.hyperopt.run import hyperopt
-from ludwig.utils.data_utils import replace_file_extension
 from tests.integration_tests.utils import category_feature, generate_data, text_feature
 
 
-@pytest.fixture(scope="session", autouse=True)
-def init_tensorflow_cpu(request):
-    """Initialize tensorflow at the start of testing to only use CPUs.
-
-    This fixture runs once before any tests, and ensures that the main process
-    running pytest does not claim any GPU resources.
-
-    This is critical to avoid OOM errors when running subprocesses that need GPUs (e.g., hyperopt),
-    as otherwise the main process will consume all the memory and cause the subprocesses to crash.
-
-    Run most tests eagerly as the cost of graph construction can easily increase runtime by
-    and order of magnitude for small tests. Tests that execute in subprocesses, and tests
-    in `test_graph_execution.py` still run in graph mode.
-    """
-    # tf.config.experimental_run_functions_eagerly(True)
-    # initialize_tensorflow(gpus=-1)
-    pass
-
-
 @pytest.fixture()
 def csv_filename():
-    """This methods returns a random filename for the tests to use for generating temporary data.
-
-    After the data is used, all the temporary data is deleted.
-    :return: None
-    """
+    """Yields a csv filename for holding temporary data."""
     with tempfile.TemporaryDirectory() as tmpdir:
         csv_filename = os.path.join(tmpdir, uuid.uuid4().hex[:10].upper() + ".csv")
         yield csv_filename
 
 
 @pytest.fixture()
-def yaml_filename():
-    """This methods returns a random filename for the tests to use for generating a config file.
+def tmpdir():
```

Comment:
This fixture already exists natively in pytest, just need to use it (no imports etc. needed):

https://docs.pytest.org/en/latest/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures

Reply:
Good to know! Very sensible that this is a built-in fixture of pytest.